sklearn/tests/test_pipeline.py

```diff
@@ -165,6 +166,27 @@ def test_pipeline_fit_params():
     assert_true(pipe.named_steps['transf'].b is None)
 
 
+def test_pipeline_raise_set_params_error():
+
+    svm  = Pipeline([('kbest', SelectKBest()),
+                     ('cls', SVC())])
+
+    pipe = Pipeline([('pca', PCA()),
+                     ('ova_svm', OneVsRestClassifier(svm))])
+    
+    # check top level parameter
+    with assert_raises(ValueError) as cm:
```

Comment:
Can you maybe add a really "top level" error for pipeline, i.e. ``set_params(fake="nope")``